plugins/bean-validation/src/main/java/org/apache/struts/beanvalidation/validation/interceptor/BeanValidationInterceptor.java

```diff
@@ -95,36 +97,42 @@ protected String doIntercept(ActionInvocation invocation) throws Exception {
         if (LOG.isDebugEnabled()) {
             LOG.debug("Validating [{}/{}] with method [{}]", invocation.getProxy().getNamespace(), invocation.getProxy().getActionName(), methodName);
         }
+        Class<?>[] validationGroup = getValidationGroups(action, methodName);
 
         if (null == MethodUtils.getAnnotation(getActionMethod(action.getClass(), methodName), SkipValidation.class,
                 true, true)) {
             // performing bean validation on action
-            performBeanValidation(action, validator);
+            performBeanValidation(action, validator, validationGroup);
         }
 
         return invocation.invoke();
     }
 
-    protected void performBeanValidation(Object action, Validator validator) {
+    protected Class<?>[] getValidationGroups(Object action, String methodName) throws NoSuchMethodException {
+        ValidateGroup validateGroup = MethodUtils.getAnnotation(getActionMethod(action.getClass(), methodName), ValidateGroup.class, true, true);
+        return validateGroup == null ? new Class[]{} : validateGroup.value();
```

Comment:
With default value defined (as proposed above with `Default.class`) this check won't be needed

Reply:
'null' check? , get NPE whithout them if no @ValidationGroup is set on ActionMethode

Reply:
@lukaszlenart meant you will not need to check if null, if you accept to change annotation as below (Currently you get NPE because you didnot accept and are discussing if `javax....` package dependencies is ok)

```java
@Retention(RetentionPolicy.RUNTIME)
public @interface ValidateGroup {

    Class<?>[] value() default {Default.class};
```

Reply:
MethodUtils.getAnnotation(...) returns 'null' if the ActionMethode has no Annotation.

Reply:
👍 you're right :) thanks! So , again , here also , to keep backward compatibility , what do you feel about following? (I'm worry because current version does not pass any group then `javax....` assumes `Default.class` then we have to keep this behavior for backward compatibility)
```java
return validateGroup == null ? new Class[]{Default.class} : validateGroup.value();
```

Reply:
👍